locations/spiders/DskBank.py

```diff
@@ -0,0 +1,67 @@
+import re
+
+import scrapy
+
+from locations.categories import Categories, apply_category
+from locations.hours import DAYS_BG, OpeningHours, sanitise_day
+from locations.items import Feature
+
+
+class DSKBankBGSpider(scrapy.Spider):
+    name = "dsk_bank_bg"
+    item_attributes = {"brand": "Банка ДСК", "brand_wikidata": "Q5206146"}
+    allowed_domains = ["https://www.dskbank.bg/"]
+    start_urls = ["https://dskbank.bg/контакти/клонова-мрежа/GetOffices/"]
+
+    def parse(self, response):
+        item = Feature()
+        response_json = response.json()
+        for data in response_json:
+            item["ref"] = data["Id"]
+            item["name"] = data["Name"]
+            item["lat"] = data["Latitude"]
+            item["lon"] = data["Longitude"]
+            item["email"] = data["Email"]
+
+            branch_type = data["BranchType"]
+            if "Branch" in branch_type:
+                apply_category(Categories.BANK, item)
+            else:
+                apply_category(Categories.ATM, item)
+
+            text = data["Text"]
+            clean_text = re.sub(r"<.*?>", "\n", text)
+            clean_text = re.sub("\n+", "\n", clean_text)
+            clean_text = clean_text.strip("\n").split("\n")
+
+            item["addr_full"] = clean_text[0]
```

Comment:
I don't know anything about Bulgarian addresses, but this is probably `street_address`, not `addr_full`.

Reply:
Thank you for pointing that out. I apologize for the confusion. I'll check this out and get back.

Reply:
I believe this could/should be replaced with: `item["addr_full"] = data["Address"]` and that `data["Text"]` (HTML data) does not need parsing. In the future, if you do need to parse HTML, use this approach:

```
from scrapy import Selector
item["addr_full"] = Selector(text=data["Text"]).xpath('//p[1]/span/text()').get()
```

Reply:
when I was going through the data I found out that sometimes `data["Address"]` contains the full address and sometimes the HTML text. So, would you like me to select the best one possible ? and If both of them contain only the street address would it be better to add the city name to make it a complete address?

Reply:
Have you got an example? I can only see text (no HTML) in the "Address" and "City" fields in source JSON data.

Reply:
If you run DictParser over the source data like this:

```
for location in response.json():
  item = DictParser.parse(location)
  yield location
```

You'll also probably need to add after the DictParser.parse call:
```
  item["street_address"] = item.pop("addr_full", None)
```

This is because DictParser by default will see an "Address" field and assume it is a full address with city, postcode, etc rather than just the street number and street name concatenated.

We also only extract fields available for addresses, and generally don't try to extract postcodes from full addresses, city names from full addresses, street numbers, etc. It is essentially up to ATP consumers to clean up the source address data, likely requiring access to non-free data sets such as government/commercial street address data sets.

Reply:
This is one piece of data I found
`{
  "Id": "d7b892b1-4d38-650e-9d59-ff0900048325",
  "ServiceId": "D0022402",
  "Name": "Албена",
  "Location": "43.3661586-28.0797264",
  "Address": "к.к. Албена",
  "CityId": "1991090806",
  "Longitude": 28.0797264,
  "Latitude": 43.3661586,
  "Text": "<p><i class=\"i loc\"></i><span>к.к. Албена, Хотел \"Laguna Garden\"</span></p>",
  "BranchType": [
    "Atm"
  ],
  "Services": [],
  "Types": null,
  "OpenHours": null,
  "VaultOpenHours": null,
  "City": "Албена, обл. Добрич",
  "RegionSfId": "59cb52b1-4d38-650e-9d59-ff0900048325",
  "CitySfId": "3bb392b1-4d38-650e-9d59-ff0900048325",
  "Signature": null,
  "Email": null,
  "Phone": null,
  "CityFriendlyName": "Албена"
}`
In this context, the `Address` field displays a portion of the complete address, specifically `к.к. Албена`. However, after applying the cleaning process using `Selector(text=data["Text"]).xpath("//p[1]/span/text()").get()`, the resulting content in the `Text` field includes the complete address, `к.к. Албена, Хотел Laguna Garden`, which, in my opinion, should be considered as the accurate address. I may be wrong because I am not familiar with how addresses look in Bulgaria.